strawberry/experimental/pydantic/object_type.py

```diff
@@ -121,7 +121,10 @@ def wrap(cls):
                     # always unset because we use default_factory instead
                     default=UNSET,
                     default_factory=defaults_into_factory(
-                        default=field.default if not field.required else UNSET,
+                        # pydantic's default and default_factory are None by default
+                        default=field.default
+                        if (not field.required) and (field.default_factory is None)
+                        else UNSET,
```

Comment:
turns out pydantic's `ModelField` doesn't use it's Undefined class to mark that the default was undefined. Thats only used in `Field`, when the user defines it. Once the model class is actually instantiated, `Field` gets turned into a `ModelField` and that Undefined distinction gets lost.

Reply:
since field.default_factory is `Optional[NoArgAnyCallable]`, if it is None and if the field is not required, 
then the default value must come from field.default